### DISCUSSION TITLE
It would be good if there was a feature to make it so that you needed to merge in recent changes when merging changes to a PR

### DISCUSSION THREAD

**Discussion by ollie-iterators** [community/community#52848] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I think it would be interesting if there was a feature to make it so that recent changes needed to be added to a PR if you were adding a new commit when there were recent changes added to the PR so that people would not push changes to a PR and miss changes that may fix tests that are failing that was merged in a different PR.

**Comment by airtower-luna** [comment]
If I understand your post correctly, setting up a branch protection rule with required checks and ["Require branches to be up to date before merging" enabled](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-status-checks-before-merging) should do what you want. :slightly_smiling_face:

**Reply by ollie-iterators** [reply]
It might be, I am thinking of having it be done as the PR is being created.

**Reply by ollie-iterators** [reply]
Would "merging" mean when the changes are merged into a pull request (but not merged into the project) as well as when a PR is merged.

**Reply by airtower-luna** [reply]
In what context? The "merging" in "Require branches to be up to date before merging" refers to merging the PR.

**Reply by ollie-iterators** [reply]
I think the change would actually be a bit earlier, like while the PR is being made.

**Reply by ollie-iterators** [reply]
I think the best way to do it would be to have it be optional or to have certain types of PR's be immune from it.
